### DISCUSSION TITLE
SVG Import Into Inkscape Issues

### DISCUSSION THREAD

**Discussion by CupcakeMunster** [netbox-community/netbox#10678] [Q&A] [answered]
If I download a cable trace svg and import that into Inkscape there are issues with how the image is displayed.

- Text is not aligned and spills out of some device boxes
- cable outline stroke becomes a fill instead 
- boxes are missing their outline

![Capture](https://user-images.githubusercontent.com/26608204/196473916-8f4638d1-2ec9-455b-a961-72713f59de70.PNG)

Am I messing something up importing the svg into other programs?

**Comment by candlerb** [comment, marked as answer]
Does the svg display properly in your browser, if you open it as a standalone file? You can use a url like `file:///path/to/file.svg`

If it does, then I'd say this is something to take up with the Inkscape project.

**Reply by CupcakeMunster** [reply]
Browser is good straight from netbox and browsing the `file:///path/to/file.svg` works too.

Seems like inkscape has known issues importing svg or other programs conform to a different standard for svg?????????
I just tried it in Visio with no issue and if I save the svg from there Inkscape has no issues importing then.

**Reply by candlerb** [reply]
> Seems like inkscape has known issues importing svg

It would seem to be so.  I suggest you raise it with the Inkscape project: if you give them your svg file, a screenshot of a browser rendering, and a screenshot of the Inkscape rendering, then there is a reproducible test case for them.

It's possible this file contains some SVG features that Inkscape doesn't support, although Inkscape does [describe itself](https://inkscape.org/about/features/) as having "Perfectly compliant SVG format file generation and editing".

**Reply by CupcakeMunster** [reply]
Yeah I'm looking at the files generated by netbox and visio and vscode linters. Every file is vastly different. I think visio and browsers are designed to be more tolerant of non-conforming svg files. There are still differences too between what visio and what browsers display but they're very minor (cables don't have black outline in visio).

I get the feeling inkscape will just tell me they are conforming to the file standard, but I'll try.

**Reply by chicks-net** [reply]
> I get the feeling inkscape will just tell me they are conforming to the file standard, but I'll try.

I think they're a bit more flexible than that.

Please paste the link to your inkscape issue over here so we can follow along.

**Reply by CupcakeMunster** [reply]
https://gitlab.com/inkscape/inkscape/-/issues/3907

**Reply by CupcakeMunster** [reply]
https://gitlab.com/inkscape/inkscape/-/issues/3909

Looks like they've identified it as a CSS variable issue with inkscape. They're tracking it as a bug now. Thanks everyone.

**Comment by jeremystretch** [comment]
IIRC, the specific issue with the black "shadow" along the cable fanout has to do with open vs. closed polygons.

**Comment by jcollie** [comment]
I noticed on the Inscape issue that @CupcakeMunster is using Inkscape 1.0. That's a pretty old version at this point. I'm sure that the Inkscape devs will ask you to reproduce the problem on the current version (1.2.1 at this point) before they look at this. When I tried opening the SVG from the Inkscape issue in Inkscape 1.2.1 I get the following:

```
$ inkscape demo.netbox.svg 
parsing error: 1:0:could not recognize next production
parsing error: 1:7:while parsing rulset: current char must be a '}'
parsing error: 1:6:while parsing declaration: next property is malformed
```

You also don't mention which version of NetBox that you used to generate the SVG, although I see the same problem in a cable trace that I generated with 3.3.5.

**Reply by CupcakeMunster** [reply]
I tried it on the latest release and reproduced it there too. I took the svg from the netbox demo site. v3.3.5
